common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -41,6 +42,9 @@
  */
 public abstract class SingleThreadEventExecutor extends AbstractScheduledEventExecutor {
 
+    protected static final int DEFAULT_MAX_PENDING_TASKS = Math.max(16,
+            SystemPropertyUtil.getInt("io.netty.eventexecutor.maxPendingTasks", 1024 * 1024));
```

Comment:
Actually we changed to 1024 * 1024 before ... I would prefer to not use Integer.MAX_VALUE honestly.